db/dbformat.cc

```diff
@@ -131,7 +131,7 @@ void InternalKeyComparator::FindShortestSeparator(
   Slice user_limit = ExtractUserKey(limit);
   std::string tmp(user_start.data(), user_start.size());
   user_comparator_->FindShortestSeparator(&tmp, user_limit);
-  if (tmp.size() < user_start.size() &&
+  if (tmp.size() <= user_start.size() &&
```

Comment:
Yes, I am also confused by this if condition. But for safety, we usually don't remove code that we are not 100% confident. This code comes from Leveldb.